### PR TITLE
Update k8s-openapi

### PR DESCRIPTION
**Pull request by olix0r**
This change updates k8s-openapi to pick up Arnavion/k8s-openapi@0a9d7e4,
which implements a workarond for a bug in the Kubernetes Node API
wherein empty lists are encoded as `null`. This workaround is necessary
until the Kubernetes v1.22 API is released including a change to the
OpenAPI spec in kubernetes/kubernetes#102159.

k8s-openapi has not yet released this fix to crates.io, so instead this
change uses a dependency patch to pull in the latest version from git.

### CODE REVIEW

**Comment by olix0r** [comment]
@clux I've marked this as a draft since these k8s-openapi changes haven't yet been released to crates.io (so we're taking a git dependency as a patch). Please let me know if there's anything I can do to be helpful moving this forward!

File: kube-core/src/resource.rs [hunk 1]
```diff
@@ -1,6 +1,5 @@
 pub use k8s_openapi::apimachinery::pkg::apis::meta::v1::ObjectMeta;
 use k8s_openapi::apimachinery::pkg::apis::meta::v1::OwnerReference;
-use once_cell::sync::Lazy;
```
Comment (kazk) [inline kube-core/src/resource.rs:3]: I believe `once_cell` is only used here, so it can be removed from `kube-core/Cargo.toml`.

**Comment by clux** [comment]
This looks great to me. Thanks a lot for doing this. Super happy with how much the metadata accessor stuff got cleaned up by this.

Did a small change in master related to dynamic objects before our 0.55 release, but don't think it should affect this branch. Looking forward to merging when k8s-openapi releases it :-)

**Comment by olix0r** [comment]
@clux k8s-openapi v0.12.0 is now on crates.io. I've updated the dependencies in this PR and I think it's ready for review/merge now. Thanks!

File: README.md [hunk 2]
```diff
@@ -1,4 +1,5 @@
 # kube-rs
+
```
Comment (olix0r) [inline README.md:2]: Uh, this must have been a markdown linter or something. I can revert these changes if desired.

**Comment by olix0r** [comment]
Note that k8s-openapi v0.12.0 supports the feature `v1_21` now. I haven't updated any of the dependencies/examples to use this API version though. Let me know if you'd prefer that I do that as part of this change.

File: README.md (hunk 2 above)
Comment (clux) [reply README.md]: ah, that's no problem, but I see you've fixed it.

**Comment by clux** [comment]
> Note that k8s-openapi v0.12.0 supports the feature `v1_21` now. I haven't updated any of the dependencies/examples to use this API version though. Let me know if you'd prefer that I do that as part of this change.

That's okay, we can wait a little with using that everywhere internally.

**Review by clux — APPROVED** [review]
Still looks great to me, I am happy to merge if CI passes!

**Comment by clux** [comment]
Released in 0.57.0. Thanks again!

**Pull request merged**